### PR TITLE
feat(stories): add initial components and scaffolding for import from…

### PR DESCRIPTION
**Pull request by cdaringe**
## WIP

- [ ] **PR for alpha review only**

### CODE REVIEW

File: .storybook/config.js [hunk 1]
```diff
@@ -1,4 +1,5 @@
-import '../lib/styles/semantic.css'
+// import '../lib/styles/semantic.css'
```
Comment (cdaringe) [inline .storybook/config.js]: i'm not sure if the build will honor sourcemaps?  given that its a dev tool only, do we want min stuffs?

File: semantic/src/site/globals/site.variables [hunk 2]
```diff
@@ -16,3 +19,20 @@
...
+@green            : #21BA45;
+@teal             : #00B5AD;
+@blue             : #3498db;
+@violet           : #6435C9;
+@purple           : #A676B2;
+@pink             : #E03997;
+@brown            : #A5673F;
+@grey             : #586A7B;
+@black            : #1B1C1D;
+@primaryColor     : @blue;
```
Comment (cdaringe) [inline semantic/src/site/globals/site.variables]: 🎉

File: src/button/Button.js [hunk 3]
```diff
@@ -1,2 +1,5 @@
 import Button from 'semantic-ui-react/dist/commonjs/elements/Button'
+
+
+
```
Comment (cdaringe) [inline src/button/Button.js]: whats the scoop?

File: src/components/FavoriteButton.jsx [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+FavoriteButton.defaultProps = {
+    isFavorited: false
+};
+
+FavoriteButton.propTypes = {
+    isFavorited: React.PropTypes.bool,
+    onClick: React.PropTypes.func,
+};
+
+export default FavoriteButton;
```
Comment (cdaringe) [inline src/components/FavoriteButton.jsx]: the build will probably barf until the linter is happy. but you may have not known that in advanced!  we use standard

File: src/components/charts/CircleChart/Circle.jsx [hunk 5]
```diff
@@ -0,0 +1,73 @@
+import React from 'react';
+import * as d3 from 'd3';
```
Comment (cdaringe) [inline src/components/charts/CircleChart/Circle.jsx]: we dont yet have tree shaking, so have pref for var `d3 = Object.assign({}, require("d3-format"), require("d3-geo"), require("d3-geo-projection"))` style imports

File: src/components/charts/CircleChart/Circle.jsx [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+  border: React.PropTypes.number,
+  endPercent: React.PropTypes.number,
+  radius: React.PropTypes.number,
+  padding: React.PropTypes.number,
+  endPercentValue: React.PropTypes.string,
+  backgroundOpacity: React.PropTypes.number,
+};
+
+CircleChart.defaultProps = {
+  color: '#4898da',
```
Comment (cdaringe) [inline src/components/charts/CircleChart/Circle.jsx]: one of our guys noted that we dont have programmatic access to our theme. we are doing stuff like this too.  a `pallette[color]` could be a cool thing before we all finish?

Comment (cdaringe) [inline src/components/charts/CircleChart/Circle.jsx]: created #4 for this!

File: src/styles/app.css [hunk 7]
```diff
@@ -0,0 +1,5 @@
+@import url('https://fonts.googleapis.com/css?family=Lato:300,400,400i,700');
```
Comment (cdaringe) [inline src/styles/app.css]: we will need to serve these locally, unfort :/

https://www.npmjs.com/package/lato-font is a thing!

File: src/styles/components/stop-start-button.css [hunk 8]
```diff
@@ -0,0 +1,15 @@
+@import '../variables.css';
+.stop-start-button {
+    cursor: pointer;
+  }
+
+  .stop-start-button:hover {
+    color: var(--blue) !important;
```
Comment (cdaringe) [inline src/styles/components/stop-start-button.css]: wait. WAT! this is a thing™??

Comment (cdaringe) [inline src/styles/components/stop-start-button.css]: we have this terrible ie11 requirement.  is there still a way?  react-scripts owns our build process

**Comment by cdaringe** [comment]
awesome, excited about this!

File: src/button/Button.js (hunk 3 above)
Comment (eddier) [inline src/button/Button.js]: I wasn't quite ready for the PR sorry about that.  Was poking around and accidentally modified the file.

File: src/components/charts/CircleChart/Circle.jsx (hunk 6 above)
Comment (eddier) [inline src/components/charts/CircleChart/Circle.jsx]: I created a colorPallete.js to deal with this.  Let me know if you want to take a different approach.

File: src/styles/components/stop-start-button.css (hunk 8 above)
Comment (eddier) [inline src/styles/components/stop-start-button.css]: Tested on IE11 to work now.

File: .storybook/config.js (hunk 1 above)
Comment (eddier) [inline .storybook/config.js]: Swapped to non-min.

File: .storybook/config.js [hunk 9]
```diff
@@ -1,4 +1,5 @@
...
+import '../semantic/dist/semantic.css';
```
Comment (jhegg) [inline .storybook/config.js]: Hi @eddier, I tried to `yarn start` on this branch, but I'm getting the same error as the CI build:
```
ERROR in ./.storybook/config.js
Module not found: Error: Cannot resolve 'file' or 'directory' ../semantic/dist/semantic.css in /data/src/octagon/.storybook
 @ ./.storybook/config.js 3:0-40
```

If I comment out line 2, and uncomment line 1 (going back to `import '../lib/styles/semantic.css'`), then it works. Did you intend to copy the `semantic.css` file somewhere?

Here are the current locations on my checkout:
```
$ find . -name semantic.css
./node_modules/semantic-ui/dist/semantic.css
./lib/styles/semantic.css
```

Comment (eddier) [inline .storybook/config.js]: Hey there - did you build the semantic ui package?  (run gulp from /semantic)

Comment (jhegg) [inline .storybook/config.js]: Oh, I did not know, sorry. Would you mind adding that into [the readme](https://github.com/Tripwire/octagon/blob/master/README.md#usage---pattern-library--storybook)?

Comment (eddier) [inline .storybook/config.js]: No problem

**Comment by cdaringe** [comment]
**[Storybook Hub](https://hub.getstorybook.io/)** is building storybooks for every commit in this PR.

**Latest Storybook: https://1833c0fe-ffcb-4327-948f-fb02a6199975.sbook.io/**
<sub>( Visit [here](https://hub.getstorybook.io/app/github.com/Tripwire/octagon/branch/slcomponents/) for more info )</sub>

> _This comment is automatically added by Storybook Hub_

File: .storybook/config.js (hunk 9 above)
Comment (jhegg) [inline .storybook/config.js]: Thanks very much!

File: src/components/NotificationItem.jsx [hunk 10]
```diff
@@ -0,0 +1,90 @@
...
+    currentTime = ''
+  }
+  if (props.notification.id === props.currentTargetNotificationId) {
+    notificationClass = 'notification__item currentTarget'
+  }
+
+  return (
+
+    <li className={` ${notificationClass}`} key={props.notification.id} >
+      <Flexbox alignItems='center' className={`notification__selection notification__state-${props.notification.state} same-timegroup-${sameTimeGroup}`}>
```
Comment (ggascoigne) [inline src/components/NotificationItem.jsx]: Can you add some guidance as to when to use the react-flexbox and when to use the semantic-ui grid?  Superficially they look like they cover much the same ground.

File: .storybook/config.js (hunk 9 above)
Comment (jhegg) [inline .storybook/config.js]: Oops, I tested it again, and it looks like gulp is not staging `dist/semantic.css`. I think it should show up in the `gulp build` output, right?

```
$ gulp build
[14:55:09] Using gulpfile /data/src/octagon/semantic/gulpfile.js
[14:55:09] Starting 'build'...
Building Semantic
[14:55:09] Starting 'build-javascript'...
Building Javascript
[14:55:09] Starting 'build-css'...
Building CSS
[14:55:09] Starting 'build-assets'...
Building assets
[14:55:09] Created: dist/components/site.js
[14:55:09] Created: dist/components/site.min.js
[14:55:09] Created: dist/components/form.js
[14:55:10] Created: dist/components/accordion.js
[14:55:10] Created: dist/components/form.min.js
[14:55:10] Created: dist/components/accordion.min.js
[14:55:10] Created: dist/components/checkbox.js
[14:55:10] Created: dist/components/checkbox.min.js
[14:55:10] Created: dist/components/dimmer.js
[14:55:10] Created: dist/components/dropdown.js
[14:55:10] Created: dist/components/dimmer.min.js
[14:55:10] Starting 'package uncompressed css'...
[14:55:10] Starting 'package compressed css'...
[14:55:10] Finished 'build-assets' after 1.49 s
[14:55:10] Finished 'package uncompressed css' after 60 ms
[14:55:10] Finished 'package compressed css' after 51 ms
[14:55:10] Finished 'build-css' after 1.5 s
[14:55:10] Created: dist/components/dropdown.min.js
[14:55:10] Created: dist/components/embed.js
[14:55:10] Created: dist/components/embed.min.js
[14:55:10] Created: dist/components/modal.js
[14:55:11] Created: dist/components/nag.js
[14:55:11] Created: dist/components/modal.min.js
[14:55:11] Created: dist/components/nag.min.js
[14:55:11] Created: dist/components/popup.js
[14:55:11] Created: dist/components/popup.min.js
[14:55:11] Created: dist/components/progress.js
[14:55:11] Created: dist/components/rating.js
[14:55:11] Created: dist/components/progress.min.js
[14:55:11] Created: dist/components/search.js
[14:55:11] Created: dist/components/rating.min.js
[14:55:11] Created: dist/components/shape.js
[14:55:11] Created: dist/components/search.min.js
[14:55:11] Created: dist/components/sidebar.js
[14:55:11] Created: dist/components/shape.min.js
[14:55:11] Created: dist/components/sticky.js
[14:55:11] Created: dist/components/sidebar.min.js
[14:55:11] Created: dist/components/tab.js
[14:55:12] Created: dist/components/sticky.min.js
[14:55:12] Created: dist/components/tab.min.js
[14:55:12] Created: dist/components/transition.js
[14:55:12] Created: dist/components/transition.min.js
[14:55:12] Created: dist/components/api.js
[14:55:12] Created: dist/components/state.js
[14:55:12] Created: dist/components/api.min.js
[14:55:12] Created: dist/components/state.min.js
[14:55:12] Created: dist/components/visibility.js
[14:55:12] Created: dist/components/visibility.min.js
[14:55:12] Starting 'package compressed js'...
[14:55:12] Starting 'package uncompressed js'...
[14:55:12] Finished 'build-javascript' after 3.04 s
[14:55:12] Finished 'build' after 3.04 s
[14:55:14] Created: dist/semantic.min.js
[14:55:14] Finished 'package compressed js' after 2.03 s
[14:55:14] Created: dist/semantic.js
[14:55:14] Finished 'package uncompressed js' after 2.03 s
```

File: src/components/NotificationItem.jsx (hunk 10 above)
Comment (eddier) [inline src/components/NotificationItem.jsx]: I could be wrong but if I remember correctly semantics grid system is based on flexbox.  I used flexbox when we original built the prototype and have stuck with it.  In practice you could probably use either and achieve the same result.   I find it a bit faster to build with than the grid approach.

File: .storybook/config.js (hunk 9 above)
Comment (eddier) [inline .storybook/config.js]: Can you retry that?  I noticed that the themes directory was in the .gitignore.  I changed that and pushed the code back up.

**Comment by cdaringe** [comment]
@eddier, let's discuss a merge strategy.  this PR keeps expanding & will be increasingly difficult to review if we dont merge soon!

may we:

- port 3rd party code out (sans sui, unless you want to own that too!)
- set precedence for subsequent PRs.  can we work towards small PRs on next wave s.t. we can review w/ greater ease?
  - i added a storybook plugin for PRs so that our UX folks can get their eyes 👀  on inbound stories!  they probably like story specific PRs

File: .storybook/config.js (hunk 9 above)
Comment (jhegg) [inline .storybook/config.js]: Ah-hah! Yep, that fixed it. Thanks! :smile:

File: package.json [hunk 11]
```diff
@@ -66,7 +76,9 @@
     "publish-major": "npm run preversion && npm version major && git push origin master --tags && npm publish",
     "check-licenses": "ripcord licenses check",
     "semantic-release": "semantic-release pre && npm publish && semantic-release post",
-    "start": "npm run storybook"
+    "start": "npm run storybook",
+    "docs-publish": "ripcord docs --publish",
+    "postpublish": "npm run docs-publish"
```
Comment (cdaringe) [inline package.json]: i'm guessing ripcord added these.  ive added #10 to fix this.  can you drop these?  sorry about that!

File: package.json [hunk 12]
```diff
@@ -41,18 +41,28 @@
...
   "dependencies": {
+    "d3": "^4.4.0",
+    "flexbox-react": "^4.1.0",
     "react": "^15.4.1",
     "react-dom": "^15.4.1",
     "semantic-ui-react": "^0.61.1"
   },
   "scripts": {
     "build-storybook": "build-storybook",
+    "static-storybook":"build-storybook -c .storybook -o .out",
```
Comment (cdaringe) [inline package.json]: CI already builds & publishes on every master build, using just `build-storybook`.  im fine w/ this if you have strong preference, but id coordially ask that you then also patch the .gitignore & update `scripts/ci-post-whatever.js`

Comment (cdaringe) [inline package.json]: oh you did patch gitignore :)

File: src/components/NotificationItem.jsx [hunk 13]
```diff
@@ -0,0 +1,90 @@
...
+const parseDate = d3.timeParse('%m-%d-%Y %H:%M:%S')
+const timeFormat = d3.timeFormat('%I')
+let currentTime = ''
+let sameTimeGroup = false
+const warningIconUri = require('../assets/warning-yellow.svg')
+const alertIconUri = require('../assets/alert-icon.svg')
+const notificationUri = require('../assets/icon_activity_notification.svg')
+
+const NotificationItem = (props) => {
+  console.log(props)
```
Comment (eddier) [inline src/components/NotificationItem.jsx]: Oops...previous linter dealt with this.

**Comment by eddier** [comment]
@cdaringe let me know what your preferences are on this.  Just to summarize this PR is going to have the most complexity due to the fact that it includes a lot of the foundation necessary for future components.  This includes:

- D3 dependencies
- React-Flexbox dependencies
- Image assets necessary 
- Tripwire SUI theme 
- Modified build process to support CSS variables in IE

From here you'll see PR's drop in size considerably and will include only the component and accompanying files/changes to css/themes.

**Comment by cdaringe** [comment]
thanks @eddier! i added some follow up commits which tidy the build/docs/lint/etc

**Pull request merged**